src/renderer/components/InstanceModal/Rename.ts

```diff
@@ -1,35 +1,58 @@
+import { LitElement, customElement, property, html, TemplateResult } from "lit-element";
 import Instance from "../../Instance";
-import instanceRenameModalTemplate from "../../templates/modals/instances/Rename.pug";
 import InstanceListStore from "../../store/InstanceListStore";
 
-export default class Rename extends HTMLDivElement {
-	private instance: Instance | null = null;
-	private oldName: string = "";
-	public constructor() {
-		super();
+@customElement("instance-rename-modal")
+export default class Rename extends LitElement {
+	protected createRenderRoot(): this { return this; }
+
+	@property({ type: Object }) public instance: Instance | null = null;
+	@property({ type: Boolean }) private isError = false;
+
+	protected render(): TemplateResult {
+		return html`
+			<div class="header">Rename</div>
+			<div class="content">
+				<p>Enter a new name for instance <strong>${this.instance?.name}</strong></p>
+				<div class="ui fluid input ${this.isError ? "error" : ""}">
+					<input id="input-rename" type="text" name="newName" placeholder="New Name" .value=${this.instance?.name ?? ""}>
+				</div>
+				${this.isError ? html`
+					<div class="ui pointing red basic label">An instance with that name already exists</div>
+				` : ""}
+			</div>
+			<div class="actions">
+				<button class="ui button cancel blue">Cancel</button>
+				<button class="ui button approve inverted green" @click=${this.handleRename}>Rename</button>
+			</div>
+		`;
+	}
+
+	private handleRename(): boolean | void {
+		const input = this.querySelector<HTMLInputElement>("#input-rename");
+		const newName = input!.value;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This assertion is unnecessary since it does not change the type of the expression.](https://app.codacy.com/manual/lukechu10/Minecraft-Box-Launcher/pullRequest?prid=5396733)